syntax = "proto3";
package YASE.DEF;

enum TexWrappingOptions {
    REPEATED = 0;
    MIRRORED_REPEATED = 1;
    CLAMP_EDGE = 2;
    CLAMP_BORDER = 3;
}

enum TexFilterOptions {
    NEAREST = 0;
    LINEAR = 1;
}

message TextureOption {
    TexWrappingOptions wrapping = 1;
    TexFilterOptions filter = 2;
}

message Texture {
    int32 id = 1;
    string name = 2;
	string category = 3;
    int32 width = 4;
    int32 height = 5;
    int32 channels = 6;
}

message LoadedTexture {
    Texture texture = 1;
    TextureOption option = 2;
}

message TextureManager {
	int32 index = 1;
    repeated Texture textures = 2;
    repeated LoadedTexture loaded = 3;
}